guides/source/association_basics.md

```diff
@@ -2724,3 +2724,156 @@ will run a query like:
 ```sql
 SELECT "vehicles".* FROM "vehicles" WHERE "vehicles"."type" IN ('Car')
 ```
+
+Delegated Types
+----------------
+
+[`Single Table Inheritance (STI)`](#single-table-inheritance-sti) works best when there is little difference between subclasses and their attributes, but includes all attributes of all subclasses I need to create a single table.
+
+This has the disadvantage that attributes that are only used by a specific subclass are added to the same table, even though they are unnecessary attributes for another subclass, which bloats a single table.
+
+In the following example, the `Comment` does not need a `subject`, but the `Message` need a `subject`, so the `Entry` includes the `subject` attribute.
+
+```ruby
+# Schema: entries[ id, type, subject, created_at, updated_at]
+class Entry < ApplicationRecord
+end
+
+class Comment < Entry
+end
+
+class Message < Entry
+end
+```
+
+`Delegated types` solves this problem.
+
+A superclass has a table that stores attributes that are shared among all subclasses.
```

Comment:
I would rewrite these 3 sentences as such:

In order to use delegated types, we have to model our data in a particular way. The requirements are as follows:


* There is a superclass that stores shared attributes among all subclasses in it's table.
* Each subclass must inherit from the super class, and will have a separate table for any additional attributes specific to it.

This eliminates the need to define attributes in a single table that are unintentionally shared among all subclasses.

Reply:
Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2751-R2756